tests/test_BaseRegTest.py

```diff
@@ -133,5 +134,5 @@ def test_train_then_test_par(self):
             self.regression_test_par(handler)
 
     def tearDown(self):
-        if self.rank == 0:
+        if self.rank == 0 and hasattr(self, "ref_file"):
```

Comment:
This is here because the skipped test will still attempt to call `tearDown`, and will fail because `self.ref_file` is undefined. There is probably a better way to do this though..